This dissertation is a coalescence of three individual projects with Salmonella being the only area that the projects have in common. Bacteriophage (phage) treatment for the reduction of multiple drug-resistant Salmonella Newport in dairy calves has been examined in our lab from a clinical disease and food safety perspective. Previously, our lab has examined the emergence of phage-resistant Salmonella as a potential consequence of phage treatment and the affect phage-resistance has on virulence in Salmonella. We generated a spontaneous mutant resistant to 4 of 5 lytic phages in our treatment regimen. This study examined the mutation that conferred the mutation of phage-resistance in our Salmonella Newport. We also examined a chicken embryo lethality assay as a model for phage therapy against Salmonella and to investigate the virulence of Salmonella strains. We also examined the application of novel statistical methods and whole-genome sequencing when investigating the prevalence of Salmonella in a multi-species animal facility. Regarding the phage-resistant mutation, we found that short-read sequencing alone was not a valid option to locate single-nucleotide polymorphisms (SNPs) that could be attributed to the phage-resistant phenotype. By using short-read and long-read sequencing with a hybrid assembly, we found an SNP in the rfbM gene that could explain a phage-resistant phenotype. Regarding the chicken embryo lethality assay, we found that use of this assay should be used cautiously with the understanding that Salmonellaâ€™s virulence and effect on survival can be very dramatic.  The third project utilized previous studies on environmental surveillance of Salmonella, with an additional year of surveillance to investigate the prevalence of Salmonella with novel statistical methods and whole-genome sequencing. Comparing supervised machine learning algorithms (logistic regression, random forest analysis, and Markov Chain Monte Carlo (MCMC)), we found that these models may be beneficial to epidemiologists investigating widespread environmental contamination. All three models found that bovine, summer, and the dairy barns and pastures were variables of importance for our study of environmental Salmonella prevalence. We found three strains of Salmonella Muenster out of ten isolates sequenced, but all isolates appear to be genetically linked and were derived from a common ancestor. Isolates within our strain cluster have less than eight SNPs, with some clusters spanning timelines up to 1493 days. These studies contained methods that have been used before but have not been applied in the areas of research of our interest. These studies are processes of applying methods of interest to our questions about Salmonella.
